pkg/shares/non_interactive_defaults.go

```diff
@@ -0,0 +1,94 @@
+package shares
+
+// FitsInSquare uses the non interactive default rules to see if messages of
+// some lengths will fit in a square of size origSquareSize starting at share
+// index cursor. See non-interactive default rules
+// https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md#non-interactive-default-rules
+func FitsInSquare(cursor, origSquareSize int, msgShareLens ...int) (bool, int) {
+	// if there are 0 messages and the cursor already fits inside the square,
+	// then we already know that everything fits in the square.
+	if len(msgShareLens) == 0 && cursor/origSquareSize <= origSquareSize {
+		return true, 0
+	}
+	firstMsgLen := 1
+	if len(msgShareLens) > 0 {
+		firstMsgLen = msgShareLens[0]
+	}
+	// here we account for padding between the contiguous and message shares
+	cursor, _ = NextAlignedPowerOfTwo(cursor, firstMsgLen, origSquareSize)
+	sharesUsed, _ := MsgSharesUsedNIDefaults(cursor, origSquareSize, msgShareLens...)
+	return cursor+sharesUsed <= origSquareSize*origSquareSize, sharesUsed
+}
+
+// MsgSharesUsedNIDefaults calculates the number of shares used by a given set
+// of messages share lengths. It follows the non-interactive default rules and
+// assumes that each msg length in msgShareLens
+func MsgSharesUsedNIDefaults(cursor, origSquareSize int, msgShareLens ...int) (int, []uint32) {
+	start := cursor
+	indexes := make([]uint32, len(msgShareLens))
+	for i, msgLen := range msgShareLens {
+		cursor, _ = NextAlignedPowerOfTwo(cursor, msgLen, origSquareSize)
```

Comment:
[Non Blocking][Question]
~Shouldn't this also check if the message is longer than the square width and have to start at a new row?~
Now that I read the documentation of `NextAlignedPowerOfTwo`, I guess the name is a bit misleading as it reflects only looking for the next aligned power of two, and not actually checking the new row rule.
I would suggest something like: `NextSharePosition(...)`, `NextShareCursor(...)` or similar

Reply:
wait, does it not check the new row rule? it returns false if it doesn't fit on that row

https://github.com/celestiaorg/celestia-app/blob/e80a11684d21a38af1f7ead3a2ed2e4922832c60/pkg/shares/non_interactive_defaults.go#L61-L63

Reply:
yes, it does. I thought it didn't at first because of the name of the function. Until I read the documentation.

Reply:
No longer relevant. Resolving.